CRM/Contribute/Tokens.php

```diff
@@ -32,10 +32,22 @@ protected function getEntityName(): string {
   }
 
   /**
-   * Get the relevant bao name.
+   * @return string
+   */
+  protected function getEntityAlias(): string {
+    return 'contrib_';
+  }
+
+  /**
+   * Get the entity name for api v4 calls.
+   *
+   * In practice this IS just ucfirst($this->GetEntityName)
+   * but declaring it seems more legible.
+   *
+   * @return string
    */
-  public function getBAOName(): string {
-    return CRM_Core_DAO_AllCoreTables::getFullName(ucfirst($this->getEntityName()));
+  protected function getApiEntityName(): string {
+    return 'Contribution';
```

Comment:
Ok, but why is `getEntityName` returning a lower-case string in the first place? To be clear, `'contribution'` is **not** the entity name; the correct answer is `'Contribution'`, Alex.

Reply:
I understand the delicacy of not wanting to break things, but IMO it's already broken if it's returning an incorrect string. `'Contribution'` isn't just the *API* entity name, it's the **canonical** entity name per `CRM_Core_DAO_AllCoreTables`.

Reply:
@colemanw yeah - I might revisit that - but I'd rather not do it in this PR as I'm not comfortable yet about whether it will have a negative impact